pandas/tseries/tools.py

```diff
@@ -271,6 +285,14 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     >>> pd.to_datetime('13000101', format='%Y%m%d', errors='coerce')
     NaT
     """
+    
+    if origin is not None:
+        from pandas.core.api import Timestamp
+        if origin == 'julian':
+            arg = arg - Timestamp(0).to_julian_date()
+        else:
+            arg = arg + Timestamp(origin).to_julian_date() - Timestamp(0).to_julian_date()
```

Comment:
Have already added a few tests case.
https://github.com/sumitbinnani/pandas/blob/9949933259f28e59b5777cf8bc46494dd29d1556/pandas/tseries/tests/test_timeseries.py#L755

Do we need a few more?